parsargs.py

```diff
@@ -0,0 +1,328 @@
+from os.path import exists
+
+# PART 1. ARGUMENTS PARSING FUNCTIONS
+
+
+def parse_args(arguments, argument_dict):
+    """
+    Looks through all input arguments and calls functions to parse them by name
+    :param arguments: a list with all input arguments (may be a direct output from sys.argv())
+    :param argument_dict: a dictionary with all possible arguments and their values by default
+    :return: a dictionary of parsed arguments and their values
+    """
+    if len(arguments) == 1 or '--help' in arguments:  # call help page
+        find_help()
+        exit()
+    used_args = []
+    file = find_fastq_file(arguments)  # finds file for filtering (a positional argument)
+    argument_dict['--output_base_name'] = file.replace('.fastq', '').split('/')[-1]
+    used_args += [file]
+    ind = 1
+    while ind + 1 != len(arguments):  # loop over all arguments except fastq file and the 1st arg (path to a script)
+        arg = arguments[ind]
+        next_value = arguments[ind + 1]
+        if arg in used_args:  # check if argument was assigned before
+            print(f'\nArgument {arg} is given more than once.\n'
+                  f'Please try again')
+            exit()
+        if arg.startswith('--'):  # check if arg is an argument, not a value
+            pass
+        else:
+            print(f'\nUnknown value used: {arg}\n'
+                  f'Please try again')
+            exit()
+        if arg not in argument_dict.keys():  # check if it is a known argument for this script
```

Comment:
```suggestion
        if arg not in argument_dict:
```
Достаточно так, комментарий в целом излишний